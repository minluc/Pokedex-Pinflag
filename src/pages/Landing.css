.landing-container {
  position: relative;
  display: flex;
  justify-content: center;
  align-items: center;
  min-height: 100vh;
  background: linear-gradient(to bottom, #c5eedb, #89b88f);
}

.landing-content {
  position: relative;
  width: 350px;
  min-height: 500px;
  background: rgba(255, 255, 255, 0.15);
  border-radius: 30px;
  border: 1px solid rgba(255, 255, 255, 0.25);
  padding: 4rem;
  text-align: center;
  backdrop-filter: blur(15px);
  -webkit-backdrop-filter: blur(15px);
  z-index: 2;
  box-shadow: 0 8px 32px rgba(0, 0, 0, 0.2);
  animation: fadeInCenter 1s ease-out forwards;
}

.landing-title {
  margin-bottom: 1rem;
}

.landing-subtitle {
  font-size: 1.1rem;
  margin-bottom: 2rem;
  color: #333;
  margin-top: 3rem;
}

.start-button {
  padding: 0.75rem 2rem;
  font-size: 1rem;
  background-color: #2f2f2f;
  color: #fff;
  border: none;
  border-radius: 30px;
  cursor: pointer;
  transition: background-color 0.3s ease, transform 0.3s ease;
}

.start-button:hover {
  background-color: #1a1a1a;
  transform: scale(1.08);
}

.charmander-image {
  position: absolute;
  left: calc(50% - 620px);
  top: 25%;
  transform: translateY(-70%);
  width: 500px;
  z-index: 10;
  filter: drop-shadow(4px 4px 12px rgba(0, 0, 0, 0.3));
  animation: slideInLeft 2s ease-out forwards;
}

.pikachu-image {
  position: absolute;
  right: calc(50% - 560px);
  top: 28%;
  transform: translateY(-70%);
  width: 360px;
  z-index: 10;
  filter: drop-shadow(4px 4px 12px rgba(0, 0, 0, 0.3));
  animation: slideInRight 2s ease-out forwards;
}

@keyframes slideInLeft {
  0% {
    transform: translateX(-100px);
    opacity: 0;
  }
  100% {
    transform: translateX(0);
    opacity: 1;
  }
}

@keyframes slideInRight {
  0% {
    transform: translateX(100px);
    opacity: 0;
  }
  100% {
    transform: translateX(0);
    opacity: 1;
  }
}

@keyframes fadeInCenter {
  0% {
    transform: scale(0.95);
    opacity: 0;
  }
  100% {
    transform: scale(1);
    opacity: 1;
  }
}

@media (max-width: 1024px) and (min-width: 769px) {
  .charmander-image {
    width: 250px;
    left: calc(50% - 400px);
  }

  .pikachu-image {
    width: 180px;
    right: calc(50% - 370px);
  }
}

@media (max-width: 768px) {
  .landing-container {
    flex-direction: column;
    padding: 1rem;
  }

  .landing-content {
    width: 90%;
    min-height: 380px;
    padding: 2rem 1.5rem;
  }

  .charmander-image {
    display: none;
  }

  .pikachu-image {
    position: static;
    width: 150px;
    margin-top: 1.5rem;
    margin-bottom: 1rem;
    transform: none;
    animation: none;
    display: block;
    align-self: center;
  }
}



