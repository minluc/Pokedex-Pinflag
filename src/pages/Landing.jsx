import { useState } from 'react';
import { useNavigate } from 'react-router-dom';
import './Landing.css';
import pikachu from '../assets/pikachu.png';
import charmander from '../assets/charmander.png';
import Loader from '../components/Loader';

function Landing() {
  const navigate = useNavigate();
  const [loading, setLoading] = useState(false);

  const handleStart = () => {
    setLoading(true);
    setTimeout(() => {
      navigate('/pokemons');
    }, 2000); 
  };

  return (
    <>
      {loading && <Loader />}

      {!loading && (
        <div className="landing-container">
          <img src="src/assets/Logo.png" className="Logo" />

          <img src={charmander} alt="Charmander" className="charmander-image" />

          <div className="landing-content">
            <h1 className="landing-title">Welcome to Pokedex!</h1>
            <p className="landing-subtitle">
              Discover, learn, and be amazed by the world of Pok√©mon. <br/><br/>
              Explore each creature, learn their abilities, and find your favorites.
            </p>
            <button className="start-button" onClick={handleStart}>
              START
            </button>
          </div>

          <img src={pikachu} alt="Pikachu waving" className="pikachu-image" />
        </div>
      )}
    </>
  );
}

export default Landing;


