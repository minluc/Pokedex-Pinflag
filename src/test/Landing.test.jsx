import { describe, it, expect, vi, beforeEach } from 'vitest'
import { render, screen, fireEvent } from '@testing-library/react'
import { BrowserRouter } from 'react-router-dom'
import Landing from '../pages/Landing'

const mockNavigate = vi.fn()
vi.mock('react-router-dom', async () => {
  const actual = await vi.importActual('react-router-dom')
  return {
    ...actual,
    useNavigate: () => mockNavigate
  }
})

vi.mock('../components/Loader', () => ({
  default: () => <div data-testid="loader">Loading...</div>
}))

describe('Landing Component', () => {
  beforeEach(() => {
    vi.clearAllMocks()
  })

  it('renders welcome message and start button', () => {
    render(<Landing />)
    expect(screen.getByText('Welcome to Pokedex!')).toBeInTheDocument()
    expect(screen.getByText('START')).toBeInTheDocument()
  })

  it('shows loader when start button is clicked', () => {
    render(<Landing />)
    fireEvent.click(screen.getByText('START'))
    expect(screen.getByTestId('loader')).toBeInTheDocument()
  })

  it('navigates to pokemons after delay', async () => {
    vi.useFakeTimers()
    render(<Landing />)
    fireEvent.click(screen.getByText('START'))

    vi.advanceTimersByTime(2000)

    expect(mockNavigate).toHaveBeenCalledWith('/pokemons')
    vi.useRealTimers()
  })

  it('displays pokemon images', () => {
    render(<Landing />)
    expect(screen.getByAltText('Pikachu waving')).toBeInTheDocument()
    expect(screen.getByAltText('Charmander')).toBeInTheDocument()
  })
})