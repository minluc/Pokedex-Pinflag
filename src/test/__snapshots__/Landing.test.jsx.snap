// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Landing Component > matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="landing-container"
  >
    <img
      class="Logo"
      src="src/assets/Logo.png"
    />
    <img
      alt="Charmander"
      class="charmander-image"
      src="charmander.png"
    />
    <div
      class="landing-content"
    >
      <h1
        class="landing-title"
      >
        ¡Bienvenido a Pokedex!
      </h1>
      <p
        class="landing-subtitle"
      >
        Descubre, aprende y sorpréndete con el mundo de los Pokémon. 
        <br />
        <br />
        Explora cada criatura, conoce sus habilidades y encuentra tus favoritos.
      </p>
      <button
        class="start-button"
      >
        START
      </button>
    </div>
    <img
      alt="Pikachu waving"
      class="pikachu-image"
      src="pikachu.png"
    />
  </div>
</DocumentFragment>
`;
